ctera/src/main/java/ch/cyberduck/core/ctera/CteraAttributesFinderFeature.java

```diff
@@ -0,0 +1,162 @@
+package ch.cyberduck.core.ctera;
+
+import ch.cyberduck.core.Acl;
+import ch.cyberduck.core.LocaleFactory;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.PathAttributes;
+import ch.cyberduck.core.dav.DAVAttributesFinderFeature;
+import ch.cyberduck.core.dav.DAVPathEncoder;
+import ch.cyberduck.core.dav.DAVSession;
+import ch.cyberduck.core.exception.AccessDeniedException;
+import ch.cyberduck.core.exception.BackgroundException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.xml.namespace.QName;
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.github.sardine.DavResource;
+
+public class CteraAttributesFinderFeature extends DAVAttributesFinderFeature {
+    private static final Logger log = LogManager.getLogger(CteraAttributesFinderFeature.class);
+
+    public static final String CTERA_GUID = "guid";
+    public static final String CTERA_NAMESPACE_URI = "http://www.ctera.com/ns";
+    public static final String CTERA_NAMESPACE_PREFIX = "ctera";
+    /**
+     * Read Data: Allows or denies viewing data in files.
+     */
+    public static final Acl.Role READPERMISSION = new Acl.Role("readpermission");
+    /**
+     * Write Data: Allows or denies making changes to a file and overwriting existing content.
+     */
+    public static final Acl.Role WRITEPERMISSION = new Acl.Role("writepermission");
+    /**
+     * Execute File: Allows or denies running program (executable) files.
+     * Files only.
+     */
+    public static final Acl.Role EXECUTEPERMISSION = new Acl.Role("executepermission");
+    /**
+     * Allows or denies deleting the file or folder. If you don't have Delete permission on a file or folder,
+     * you can still delete it if you have been granted Delete Subfolders and Files on the parent folder.
+     */
+    public static final Acl.Role DELETEPERMISSION = new Acl.Role("deletepermission");
+    /**
+     * Create Files: Allows or denies creating files within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEFILEPERMISSION = new Acl.Role(WRITEPERMISSION.getName());
```

Comment:
@dkocher @ylangisc should we distinguish `createfilepermission` in Cyberduck core from `writepermission` at all, or discard `createfilepermission`?

Reply:
I left the definition in place to keep existing preflight implementations that distinguish which will allow possible future use.